### PR TITLE
doc: more clear statement about code splitting

### PR DESCRIPTION
**Pull request by theKashey**
Related to https://github.com/gaearon/react-hot-loader/issues/947
A bit more clear statement about code splitting..

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,25 +203,36 @@ Using React Hot Loader with React Native can cause unexpected issues (see #824)
...
+import asyncComponent from 'react-async-component'
+const AsyncHello = asyncComponent({
+  resolve: () => import('./Hello.js'),
+})
 
 // Hello.js
 import { hot } from 'react-hot-loader'
 const Hello = () => 'Hello'
-export default hot(module)(Hello) // <-- the only change to do
+export default hot(module)(Hello) // <-- module will reload itself
```
Comment (neoziro) [inline README.md]: You should use `loadable-components` or `react-imported-component`. They provide the best support for RHL.

File: README.md [hunk 2]
```diff
@@ -203,25 +203,36 @@ Using React Hot Loader with React Native can cause unexpected issues (see #824)
 
 ### Code Splitting
 
-Most of modern React component-loader libraries ([loadable-components](https://github.com/smooth-code/loadable-components/),
-[react-loadable](https://github.com/thejameskyle/react-loadable)...) are compatible with React Hot Loader.
+Most of modern React component-loader libraries are not "100%" compatible with React-Hot-Loader, as long
```
Comment (neoziro) [inline README.md]: We should discourage using `react-loadable`.

**Comment by neoziro** [comment]
I made some test on Loadable Components. I don't know what has changed but I no longer need to reload everything. It works out of the box without adding any code. Could be the same for React Loadable.

**Comment by theKashey** [comment]
You know, it sounds like a bug. Ok a huge change inside webpack. 
It could only work is something triggered “register”. 
Can you share that example?

**Comment by neoziro** [comment]
Example from [this PR](https://github.com/smooth-code/loadable-components/pull/73). Let's make it clear before rewriting this README section.

**Comment by theKashey** [comment]
@neoziro - so everything is due to autoload inside the constructor. Look like the condition should be a bit different, - preload only NOT for browser, to emulate `componentWillMount` on server-side.
In [imported-component](https://github.com/theKashey/react-imported-component/blob/master/src/Component.js#L36) I am using `detect-node` for it. Fix and test this moment just last weekends, so I am pretty sure you shall do the same.

As result - the state of code splitting is unchanged. You still have somehow trigger reimport, and it will not work without it.

**Comment by neoziro** [comment]
@theKashey the behaviour that I want is loading in component but only server-side https://github.com/smooth-code/loadable-components/blob/master/src/loadable.js#L58. So I think doing it in constructor makes it compatible with HMR, so nice!

Yes the state of Code Splitting stay unchanged. Loadable Components and Imported Components work both out of the box.

**Comment by theKashey** [comment]
But `typeof window !== 'undefined'` is a browser, not server side.
Anyway, this one is good to merge.

**Pull request merged**